### PR TITLE
lib: src: simulations: Added Maurer Rose Pattern

### PR DESCRIPTION
**Pull request by kmrakash**

### CODE REVIEW

**Comment by kmrakash** [comment]
It's Working FIne

File: lib/src/data/simulations.dart [hunk 1]
```diff
@@ -25,6 +26,7 @@ class Simulations with ChangeNotifier {
     4: "fourier series mathematics ",
     5: "lissajous curve pattern mathematics animation ",
     6: "epicycloid curve pattern mathematics animation pencil lines ",
+    7: "moure rose pattern"
```
Comment (shreyamalviya) [inline lib/src/data/simulations.dart]: @kmrakash, just scanned this for now and I noticed that the spelling is wrong at a _lot_ of places in the PR, so you should probably fix that and keep it in mind for next time. It's "Maurer". 

Great effort, by the way!

**Review by shreyamalviya — CHANGES_REQUESTED** [review]
Remove ".svg" from the filename

**Comment by kmrakash** [comment]
.svg is mandatory for this file otherwise figure disappear . I tried it

File: lib/src/simulations/mourer_rose.dart [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+                  setState(() {
+                    _n = double.parse(value.toStringAsFixed(1));
+                  });
+                },
+                value: _n,
+              ),
+              Center(
+                child: Text(
+                  "N: $_n",
+                 // style: Theme.epicycloid_curveof(context).textTheme.subtitle,
```
Comment (shreyamalviya) [inline lib/src/simulations/mourer_rose.dart]: stray comment

File: lib/src/data/simulations.dart [hunk 3]
```diff
@@ -25,6 +26,7 @@ class Simulations with ChangeNotifier {
     4: "fourier series mathematics ",
     5: "lissajous curve pattern mathematics animation ",
     6: "epicycloid curve pattern mathematics animation pencil lines ",
+    7: "mourer rose pattern"
```
Comment (shreyamalviya) [inline lib/src/data/simulations.dart]: spelling

**Review by shreyamalviya — CHANGES_REQUESTED** [review]
Again, the spelling is "Maurer". Please fix it at _all_ the places where it is wrong.
Other than that, LGTM.

@yashlamba, can you make the final image for this one?

**Comment by yashlamba** [comment]
@sakshamarora1 and @yashugarg can help review this further!?

**Comment by kmrakash** [comment]
Oh sorry it should be "Maurer", my mistake ,
Anything else i have to made changes please let me know . 
Before this my only motive was to function properly . I have not care that much on spellings . Let me correct these
Thanx for review

**Review by sakshamarora1 — CHANGES_REQUESTED** [review]
It looks really good!
Just a little tweaks and I think it's good to go!

File: lib/src/simulations/maurer_rose.dart [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+}
+
+class MaurerRosePainter extends CustomPainter {
+  double d, n, c;
+  double k, transformx, transformy;
+  List<Offset> points = [];
+  List<Offset> points2 = [];
+  bool animate;
+  double thickness;
+  MaurerRosePainter(
```
Comment (sakshamarora1) [inline lib/src/simulations/maurer_rose.dart:370]: How about passing a color parameter and then keeping the color of inner lines opposite to the theme of the app.

**Review by sakshamarora1 — CHANGES_REQUESTED** [review]
It looks really good!
Just a little tweaks and I think it's good to go!

File: lib/src/simulations/maurer_rose.dart [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.animating) {
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        nextStep();
+      });
+    }
+    return CustomPaint(
+      painter: MaurerRosePainter(
```
Comment (sakshamarora1) [inline lib/src/simulations/maurer_rose.dart]: Just pass ``Theme.of(context).accentColor`` and assign color to paint.color in the CustomPainter

File: lib/src/simulations/maurer_rose.dart [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+    ..style = PaintingStyle.stroke
+    ..strokeWidth = thickness;
+    if (!animate) {
+      this.points.clear();
+      for (var theta=0; theta <= 360; theta++) {
+        var k = theta*d*pi/180;
+        var q= 300*sin(n*k);
+        var x = q*cos(k) ;
+        var y = q*sin(k) ;
+        this.points.add(Offset(x/2,y/2)
```
Comment (sakshamarora1) [inline lib/src/simulations/maurer_rose.dart]: The curve is not filling up the screen completely so how about changing this to be divided by 1.5
same goes for all other instances of this

File: lib/src/simulations/maurer_rose.dart [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+    this.points2 = new List<Offset>.from(points2);
+    k = n / d;
+  }
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    var paint = Paint()
+    ..color = Colors.black
+    ..style = PaintingStyle.stroke
+    ..strokeWidth = thickness;
```
Comment (sakshamarora1) [inline lib/src/simulations/maurer_rose.dart]: There's no need to keep the thickness of the inner lines the same as the outer curve. let's just keep it constant to 1?

**Comment by kmrakash** [comment]
@sakshamarora1  now it is showing some conflicts , How can i resolve ?? Please Help

**Comment by sakshamarora1** [comment]
@kmrakash No worries, I've resolved it. Thank you!
There was a conflict because another PR got merged recently.

**Comment by kmrakash** [comment]
Thank you @sakshamarora1 , will it merge now ??

**Review by yashlamba — APPROVED** [review]
This looks good to merge, just make sure to remove the svg file before merging @sakshamarora1. Good work @kmrakash.

**Pull request merged**